### PR TITLE
REST API: Posts: Enable querying by custom taxonomy terms

### PR DESCRIPTION
**Pull request by zinigor**
Previously had only been possible to query by category or tag

Merges r140311-wpcom.

### CODE REVIEW

File: json-endpoints/class.wpcom-json-api-list-posts-endpoint.php [hunk 1]
```diff
@@ -172,6 +172,40 @@ function callback( $path = '', $blog_id = 0 ) {
...
+
+				$query['tax_query'][] = array(
+					'taxonomy' => $taxonomy,
+					'field' => 'slug',
+					'terms' => explode( ',', $slug )
+				);				
+			}
+		}
+
+		if ( ! empty( $args['term'] ) ) {
```
Comment (aduth) [inline json-endpoints/class.wpcom-json-api-list-posts-endpoint.php]: Looks like this was duplicated?

Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-list-posts-endpoint.php]: Yes, you're right, I'll fix that, thank you!

**Comment by zinigor** [comment]
@aduth I have fixed the duplicated code, thanks for taking a look!

**Comment by zinigor** [comment]
Master issue: #7026

**Comment by eliorivero** [comment]
I tested this on my site adding a custom taxonomy and hitting this in api console
`/sites/DOTORGSANDBOX/posts/?term=testslug`
and it's returning nothing. I do have a post with that term.

**Comment by aduth** [comment]
@eliorivero The correct parameter would be `?term[post_tag]=testslug` if it's a tag you're trying to query.

**Comment by eliorivero** [comment]
Great @aduth! it works with that syntax, except I had created a tax so it's like
`/sites/DOTORGSANDBOX/posts/?term[testaxonomy]=testslug`
Thanks!

**Pull request merged**